File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/managers/events/IEventDispatcher.java
package com.dkellycollins.triggerfinger.managers.events;

/**
 * Provides the ability to broadcast event data.
 */
public interface IEventDispatcher {

    /**
     * Broadcasts an event to the appropriate interceptors.
     * @param event The event data to send.
     * @param <T> The type of the event.
     */
    <T extends IEvent> void dispatch(T event);
}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/managers/events/IEventDispatcher.kt
package com.dkellycollins.triggerfinger.managers.events

/**
 * Provides the ability to broadcast event data.
 */
interface IEventDispatcher {

    /**
     * Broadcasts an event to the appropriate interceptors.
     * @param event The event data to send.
     * *
     * @param  The type of the event.
     */
    fun <T : IEvent> dispatch(event: T)
}
